### ISSUE TITLE
query parameter caught by substring query parameters

### ISSUE THREAD

**Issue by thirsty-wing** [elysiajs/elysia#485]
### What version of Elysia.JS is running?

0.8.17

### What platform is your computer?

Darwin 23.2.0 arm64 arm

### What steps can reproduce the bug?

Hitting this endpoint `/what?xxltee=on` with this code reproduces the problem

```
import { Elysia, t } from "elysia";
import { html } from "@elysiajs/html";
import { swagger } from "@elysiajs/swagger";
import * as Html from "@kitajs/html";

const app = new Elysia()
  .use(html())
  .use(swagger())
  .get(
    "/what",
    ({ query }) => {
      console.log("query", query);
      return <h1>hi</h1>;
    },
    {
      query: t.Object({
        stee: t.Optional(t.Literal("on")),
        mtee: t.Optional(t.Literal("on")),
        ltee: t.Optional(t.Literal("on")),
        xltee: t.Optional(t.Literal("on")),
        xxltee: t.Optional(t.Literal("on")),
        xxxltee: t.Optional(t.Literal("on")),
      }),
    }
  )
  .listen(3000);

console.log(
  `🦊 Elysia is running at ${app.server?.hostname}:${app.server?.port}`
);
```

### What is the expected behavior?

I would expect  the following output when hitting the route `/what?xxltee=on`:

```
query {
  stee: undefined,
  mtee: undefined,
  ltee: undefined,
  xltee: undefined,
  xxltee: "on",
  xxxltee: undefined,
}
```

### What do you see instead?

Instead, I see the following output when hitting the route `/what?xxltee=on`:

```
query {
  stee: undefined,
  mtee: undefined,
  ltee: "on",
  xltee: "on",
  xxltee: "on",
  xxxltee: undefined,
}
```

where `ltee` and `xltee` are also "on"

**Comment by SaltyAom** [comment]
Fixed in [711925a](https://github.com/elysiajs/elysia/pull/465/commits/711925a5b5561452b58979b26917cb13a6cf8d9c) under `1.0.0-beta.14`.

If resolved, please leave the issue open until release on next stable release.

**Comment by SaltyAom** [comment]
Should be fixed in 1.0.0

Feels free to reopen if issue still persists.
